xbmc/windowing/WinSystem.h

```diff
@@ -120,6 +120,19 @@ class CWinSystemBase
   virtual bool IsTextInputEnabled() { return false; }
 
   std::string GetClipboardText(void);
+  
+  /**
+   * Get VADisplay that can be used for hardware decoding
+   * 
+   * Kodi currently does not use libva backend-specific interfaces for presenting
+   * the decoded frames.
+   * If this changes, the API for interacting with the windowing system
+   * will have to be augmented.
+   * 
+   * \return new VADisplay that can be used for hardware decoding or nullptr
+   *         if unsupported or creation failed
+   */
+  virtual void* GetVaDisplay() { return nullptr; }
```

Comment:
If this is an intermediate solution, it is ok. You just have to be aware that this can't be brought to mainline this way. Maybe add a comment with a ref to #11